src/quo/components/wallet/confirmation_progress/view.cljs

```diff
@@ -0,0 +1,78 @@
+(ns quo.components.wallet.confirmation-progress.view
+  (:require [quo.components.wallet.confirmation-progress.style :as style]
+            [quo.components.wallet.progress-bar.view :as progress-box]
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]))
+
+(defn- calculate-box-state
+  [state counter index]
+  (cond
+    (and (= state :sending) (>= counter index) (< index 3))                 :confirmed
```

Comment:
since we are making so many uses of these various states in this component I wonder is it better to define these keywords in constant variables? 🤔 
e.g `(def ^:private confirmed :confirmed)`

Reply:
@vkjr @smohamedjavid any thoughts?

Reply:
I agree. Done

Reply:
hmm I don't think so because I understand that the different components have the same names etc, but this is best to be within the component to avoid unnecessary coupling of individual ui components

Reply:
> since we are making so many uses of these various states in this component I wonder is it better to define these keywords in constant variables? 🤔 e.g `(def ^:private confirmed :confirmed)`

I always felt that the keywords (for states and types of the component) were prone to typos and would break the component if not handled correctly. We should start discussing with the team on this.

Placing the types and states constant is the best way (at least on JS), but these constants (`quo.components.CATEGORY.COMPONENT.constants`) should be exported to use on the screens. Additionally, we have a lint rule to use components only via `quo.core` (not allowing direct component imports).

Reply:
There was an issue using 'case' with this approach- any idea how to get around that? Or just don't use case? 😅

Reply:
@J-Son89 
> e.g (def ^:private confirmed :confirmed)

I think that's not the best approach, since `:confirmed` is already a constant in Clojure. IMO this is exactly the problem keywords solve:

instead of:
`(def ^:const confirmed "confirmed")`
just:
`:confirmed`
They are already constants, they are being cached, and (AFAIK) multiple uses of the a keyword point to the same keyword.

@smohamedjavid 
> I always felt that the keywords (for states and types of the component) were prone to typos and would break the component if not handled correctly

I haven't had a typo in a keyword, probably because the code editor suggest the existing ones.
Sure, it'd be something to discuss.

@J-Son89 
> There was an issue using 'case' with this approach- any idea how to get around that? Or just don't use case?

Yep, case works only with literals, from the docstring:
> The test-constants are not evaluated. They must be compile-time literals, and need not be quoted.

![image](https://github.com/status-im/status-mobile/assets/90291778/6674865c-6f14-4e69-8ddd-94dd928e60cc)